protoc_plugin/lib/names.dart

```diff
@@ -70,16 +70,24 @@ class OneofNames {
       this.whichOneofMethodName, this.oneofEnumName, this.byTagMapName);
 }
 
+// For performance reasons, use code units instead of Regex.
+bool _startsWithDigit(String input) =>
+    input.isNotEmpty && (input.codeUnitAt(0) ^ 0x30) <= 9;
+
 /// Move any initial underscores in [input] to the end.
 ///
 /// According to the spec identifiers cannot start with _, but it seems to be
-/// accepted by protoc.
+/// accepted by protoc. These identifiers are private in Dart, so they have to
+/// be transformed.
 ///
-/// These identifiers are private in Dart, so they have to be transformed.
+/// If [input] starts with a digit after transformation, prefix with an 'x'.
```

Comment:
Just a thought: maybe it would be prettier to not move the underscore to the end when there is also a digit, we anyway prefix it with an x.